src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -901,9 +901,45 @@ short Hole::mustExecute() const
 
 void Hole::Restore(Base::XMLReader &reader)
 {
-    ProfileBased::Restore(reader);
-
-    updateProps();
+    reader.readElement("Properties");
+    int Cnt = reader.getAttributeAsInteger("Count");
+
+    for (int i=0 ;i<Cnt ;i++) {
+        reader.readElement("Property");
+        const char* PropName = reader.getAttribute("name");
+        const char* TypeName = reader.getAttribute("type");
+        App::Property* prop = getPropertyByName(PropName);
+
+        try {
+            if (prop && strcmp(prop->getTypeId().getName(), TypeName) == 0) {
+                prop->Restore(reader);
+            }
+            else if (prop && strcmp(TypeName,"App::PropertyFloatConstraint") == 0 &&
+                     strcmp(prop->getTypeId().getName(), "App::PropertyQuantityConstraint") == 0) {
+                App::PropertyFloatConstraint p;
+                p.Restore(reader);
+                static_cast<App::PropertyQuantityConstraint*>(prop)->setValue(p.getValue());
+            }
+        }
+        catch (const Base::XMLParseException&) {
+            throw; // re-throw
+        }
+        catch (const Base::Exception &e) {
+            std::string errorstrbase = "Thread size out of range";
+            std::string errorstr = e.what();
+            if (errorstrbase.compare(errorstr) == 0) {
+                Base::Console().Log("%s in Hole Feature can be ignored on opening an existing file\n", e.what());
+            }
+            else {
+                Base::Console().Error("%s\n", e.what());
+            }
+        }
+        catch (const std::exception &e) {
+            Base::Console().Error("%s\n", e.what());
+        }
+        reader.readEndElement("Property");
+    }
+    reader.readEndElement("Properties");

```

Comment:
It seems like large portions of this new code is taken directly from [ProfileBased::Restore](https://github.com/FreeCAD/FreeCAD/blob/81956525f6b0af04f1960cd727b6e8b13af08f6d/src/Mod/PartDesign/App/FeatureSketchBased.cpp#L1158)

Is there any way to refactor ProfileBased::Restore a bit in order to minimize/eliminate the code duplication?

Where else is ProfileBased::Restore used?

Reply:
> Where else is ProfileBased::Restore used?

This function is used by any sub-class that doesn't override the Restore() method.

Another problem of this implementation is that the string "Thread size out of range" must match with that of updateDiameterParam() and this is nowhere explicitly mentioned.
So, if someone changes the text the error message might be back because Restore() won't filter it any more.

A cleaner way is avoid to even throw the exception when we know that during the restore the feature is in an inconsiatent state: 30bc9fc6b